src/background/feeder/e2e-session.feeder.ts

```diff
@@ -0,0 +1,194 @@
+type BzzLinkLinkUrlToFakeUrlRequest = Request<
+  Action.BZZ_LINK_LINK_URL_TO_FAKE_URL,
+  { bzzLinkUrl: string; newPage: boolean }
+>
+type BzzLinkUrlToFakeUrlRequest = Request<Action.BZZ_LINK_URL_TO_FAKE_URL, { url: string; newPage: boolean }>
+type Web2HelperFakeBeeApiAddressRequesst = Request<Action.WEB2_HELPER_FAKE_BEE_API_ADDRESS, void>
+type Web2HelprFakeBzzAddressRequest = Request<Action.WEB2_HELPER_FAKE_BZZ_ADDRESS, { reference: string }>
+
+type RequestType =
+  | RegisterRequest
+  | LocalStorageGetRequest
+  | LocalStorageSetRequest
+  | BzzLinkProtocolToFakeUrlRequest
+  | BzzLinkLinkUrlToFakeUrlRequest
+  | BzzLinkUrlToFakeUrlRequest
+  | Web2HelperFakeBeeApiAddressRequesst
+  | Web2HelprFakeBzzAddressRequest
+
+export class E2ESessionFeeder {
+  constructor(private manager: DappSessionManager) {
+    this.serveEvents()
+  }
+
+  serveEvents(): void {
+    console.log('Register E2ESessionFeeder event listeners...')
+
+    // register dapp session id
+    chrome.runtime.onMessageExternal.addListener(
+      async (
+        request: RequestType,
+        sender: chrome.runtime.MessageSender,
+        sendResponse: (response?: Response) => void,
+      ) => {
+        const { action } = request || {}
+        const senderId = sender.id as string
+        const response: Response = {}
+
+        try {
+          if (!senderId) {
+            throw new Error('Invalid extension ID')
+          }
+
+          if (action === Action.REGISTER) {
+            response.data = this.handleRegistration(sender)
+          } else if (action === Action.LOCAL_STORAGE_GET) {
+            response.data = await this.handleLocalStorageGet(request)
+          } else if (action === Action.LOCAL_STORAGE_SET) {
+            await this.handleLocalStorageSet(request)
+          } else if (action === Action.WEB2_HELPER_FAKE_BEE_API_ADDRESS) {
+            response.data = this.handleWeb2FakeBeeApiAddress(request)
+          } else if (action === Action.WEB2_HELPER_FAKE_BZZ_ADDRESS) {
+            response.data = this.handleWeb2FakeBzzAddress(request)
+          } else if (action === Action.BZZ_LINK_PROTOCOL_TO_FAKE_URL) {
+            response.data = this.handleBzzLinkProtocolToFakeUrl(request)
+          } else if (action === Action.BZZ_LINK_LINK_URL_TO_FAKE_URL) {
+            response.data = this.handleBzzLinkUrlToFakeUrl(request)
+          } else if (action === Action.BZZ_LINK_URL_TO_FAKE_URL) {
+            response.data = this.handleBzzUrlToFakeUrl(request)
+          } else {
+            throw new Error(`Unknown action ${action}`)
+          }
+        } catch (error) {
+          response.error = String(error)
+        }
+
+        this.sendResponse(response, action, senderId, sendResponse)
+      },
+    )
+  }
+
+  private sendResponse(
+    response: Response | null,
+    action: Action,
+    senderId: string,
+    sendResponse: (response?: Response) => void,
```

Comment:
find different naming for this one please because it may seem a bit confusing the method itself has the same name. Could be `responseHandler` or smth similar.

Reply:
Good point. I've renamed it to 'handleResponse'.